converters/PGNConverter.cpp

```diff
@@ -105,9 +113,41 @@ int PGNConverter::getAlgebraicMoveType(std::string algebraicString) {
         return PAWN_MOVE;
     }
 
+    std::regex pawnPromotionRegEx("=[NBQR]");
+    if (std::regex_search(algebraicString, sm, pawnPromotionRegEx))
```

Comment:
This will never be reached, PAWN_MOVE already checks for `(=[NBQR])?` in it's regex

Reply:
You can do another regex under PAWN_MOVE to check if it's promoted and if it is then you can call you promotion function after the move.